src/components/ShoppingCart/ShoppingCartTable/ShoppingCartCard/ShoppingCartCard.component.jsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react'
+import { useDispatch, useSelector } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import {shoppingCartActionCreators} from 'state/index';
+import PropTypes from 'prop-types';
+
+const ShoppingCartCardComponent = ({className, data: { id, name, url, price, quantity2, stock } }) => {
+    
+    const { items } = useSelector((state) => state.shoppingCart);
+    const {quantity} = items.find((item) => item.id === id) || { quantity: 0};
+
+    const dispatch = useDispatch();
+    const { removeFromCart, changeQuantity } = bindActionCreators(shoppingCartActionCreators, dispatch);
+
+
+    const quantityOptions = Array.from(Array(stock).keys());
+    
+    return (
+        <div className={className}>
+            <img src={url} alt="caca"></img>
+            <span>{name}</span>
+            <span>Price: ${price}</span>
+            <span>Qty: 
+                <select defaultValue={quantity} onInput={(e) => changeQuantity(id, parseInt(e.target.value))}>
```

Comment:
I think this should be moved on to its own function and there you could also update the subtotal
```suggestion
                <select defaultValue={quantity} onInput={ handleQuantityonInput }>
```